### PR TITLE
tvheadend: add feature uriparser & liburiparser: add package & tvheadend: update to 2024-06-28

### PR DESCRIPTION
**Pull request by sKyissKy**
Maintainer: @sKyissKy & @M95D 
Compile tested: Ubuntu 22.04.4 LTS x86-64
Run tested: x86-64 & mvebu, Linksys WRT-1900ACv1

Description:
This is based on original pull request by @sKyissKy : https://github.com/openwrt/packages/pull/20504 and @M95D https://github.com/openwrt/packages/pull/21356

Added liburiparser.
Added tvheadend dependency on liburiparser. @M95D made it configurable by user, since most users won't need this. Default is off.
Update to tvheadend git version to commit d2e41b5 (date: 2024-06-28)

Tested:
Tvheadend works when built with the new lib option.

### CODE REVIEW

**Comment by BKPepe** [comment]
1. In your commit to re-enable uriparser for tvheadend, there is missing commit description. 
Please take a look at https://openwrt.org/submitting-patches

2. There was attempt to add liburiparser to this repository, feedback was given in https://github.com/openwrt/packages/pull/21356/files, but I dont see that it was addressed here.

**Comment by sKyissKy** [comment]
> 1. In your commit to re-enable uriparser for tvheadend, there is missing commit description.
>    Please take a look at https://openwrt.org/submitting-patches
> 2. There was attempt to add liburiparser to this repository, feedback was given in https://github.com/openwrt/packages/pull/21356/files, but I dont see that it was addressed here.

Thanks for your reminder, I have made the changes.

File: multimedia/tvheadend/Makefile [hunk 1]
```diff
@@ -33,8 +33,10 @@ define Package/tvheadend
 	+librt \
 	+libffi \
 	+libopenssl \
+	+liburiparser \
```
Comment (M95D) [inline multimedia/tvheadend/Makefile:36]: This would add an unconditional dependency on liburiparser.

**Review by M95D — COMMENTED** [review]
This would add an unconditional dependency on liburiparser.

File: multimedia/tvheadend/Makefile (hunk 1 above)
Comment (sKyissKy) [reply multimedia/tvheadend/Makefile]: Thanks for your reminder, I have made the changes.

**Comment by BKPepe** [comment]
I haven't told you that you should merge liburiparser package and enabling it for tvheadend in one single commit. :-(

Still my feedback from the other PR was not addressed, still no go, unfortunately.

File: multimedia/tvheadend/Config.in [hunk 2]
```diff
@@ -43,6 +43,12 @@ choice
       Use more advanced Perl-Compatible Regular Expressions, provided by libpcre2.
 endchoice
 
+config TVHEADEND_URIPARSER
+  bool "Use liburiparser"
+  depends on PACKAGE_tvheadend
+  select PACKAGE_liburiparser
+  default n
```
Comment (BKPepe) [inline multimedia/tvheadend/Config.in:50]: Now, this is the only config, which does not have description.

**Review by BKPepe — CHANGES_REQUESTED** [review]

**Comment by sKyissKy** [comment]
> I haven't told you that you should merge liburiparser package and enabling it for tvheadend in one single commit. :-(
> 
> Still my feedback from the other PR was not addressed, still no go, unfortunately.

Thank you for your feedback. I apologize for any misunderstanding. As per your previous feedback, I have now combined the liburiparser package and enabled it for tvheadend in a single commit. If there are any other issues or specific changes you would like to see, please let me know, and I will address them promptly.

File: multimedia/tvheadend/Config.in (hunk 2 above)
Comment (sKyissKy) [reply multimedia/tvheadend/Config.in:50]: fixed.

**Review by sKyissKy — COMMENTED** [review]
fixed

**Review by BKPepe — CHANGES_REQUESTED** [review]
1. Feedback for liburiparser package in this PR: https://github.com/openwrt/packages/pull/21356 was not addressed still.

2. Adding new package should be done in separate commit as it was done previously.

3. Update could be done also in separate PR to have it merged faster.

**Pull request closed without merging**